### PR TITLE
Add "options" argument to QuantinuumBackend constructor.

### LINKED ISSUE #33: [feature request] Allow error params to be passed in backnd constructor
These would then be used by process_circuits) by default rather than having them passed in each time.

### PR DESCRIPTION
**Pull request by cqc-alec**
Closes #33 .

### CODE REVIEW

File: pytket/extensions/quantinuum/backends/quantinuum.py [hunk 1]
```diff
@@ -454,6 +463,7 @@ def submit_qasm(
 
         if options is not None:
             body["options"].update(options)
+        body["options"].update(self._process_circuits_options)
```
Comment (cqc-melf) [inline pytket/extensions/quantinuum/backends/quantinuum.py:466]: From my understanding this allows the user to override the previously set options. Is that intended?

File: pytket/extensions/quantinuum/backends/quantinuum.py [hunk 2]
```diff
@@ -158,6 +158,7 @@ def __init__(
         provider: Optional[str] = None,
         machine_debug: bool = False,
         api_handler: QuantinuumAPI = DEFAULT_API_HANDLER,
+        **kwargs: QuumKwargTypes,
```
Comment (cqc-melf) [inline pytket/extensions/quantinuum/backends/quantinuum.py:161]: Is this a good name for the variable?

File: pytket/extensions/quantinuum/backends/quantinuum.py (hunk 1 above)
Comment (cqc-alec) [reply pytket/extensions/quantinuum/backends/quantinuum.py:466]: Yes, but I think I actually have it the wrong way round: the options specified in the backend constructor should be overridden (potentially) by the options passed to `process_circuits()`. I will change this.

File: pytket/extensions/quantinuum/backends/quantinuum.py (hunk 2 above)
Comment (cqc-alec) [reply pytket/extensions/quantinuum/backends/quantinuum.py:161]: It's just copied from the `process_circuits()` signature.

File: pytket/extensions/quantinuum/backends/quantinuum.py (hunk 1 above)
Comment (cqc-melf) [reply pytket/extensions/quantinuum/backends/quantinuum.py]: Perfect, thank you!

**Review by cqc-melf — APPROVED** [review]

**Pull request merged**